src/components/datagrid/DataGrid.js

```diff
@@ -458,7 +458,7 @@ export default class DataGridComponent extends NestedArrayComponent {
       const component = this.createComponent(columnComponent, options, row);
       component.parentDisabled = !!this.disabled;
       component.rowIndex = rowIndex;
-      component.inDataGrid = true;
```

Comment:
We should not be mutating the "component" object.  This is what causes infinite loop errors with the Logic system because the "component" changed. Why is this necessary?

Reply:
The option to 'Show Label in DataGrid' was not available at builder because the custom conditional didn't come true.
`inDataGrid` just wasn't assigned to `editComponent`.
![image](https://user-images.githubusercontent.com/54106407/103090469-2041e380-4602-11eb-9204-25854669809d.png)

Reply:
Will find another way to pass it